src/reducers/index.js

```diff
@@ -105,3 +111,20 @@ export default function reducer(state = initialState, action) {
     return newState
   }
 }
+
+function analyzeChanges(changed, submodelBasePath, attribute) {
+  if (changed == undefined){
+    if (submodelBasePath._savedAttributes[attribute] == undefined)
```

Comment:
Achtung, in JS `null == undefined // => true`. Ist das bei dieser Zeile gewollt? Ansonsten müsste hier `=== undefined` stehen.

Reply:
Hmm habe ich nicht bedacht und kommt wahrscheinlich nicht vor, sollte aber so okay sein. Wenn changed null ist, kann er ja ruhig changed definieren, finde ich - oder was meinst du? Die Abfrage soll vor allem verhindern, dass changed überschrieben wird, wenn es von außen reingegeben wird und ein null als changed reinzugeben macht in meinem Kopf nicht viel Sinn.

Reply:
Ah sry du hast den Kommentar ja auch an die Zeile gesetzt. Hmm beim coden stand da immer undefined und niemals null in den savedChanges und wenn der Key gar nicht drin steht ist das result ja undefined. Die optionalen Werte (also in ruby nil) landen ja auch als undefined in den savedChanges, oder? Und wenn da mal null drinsteht wäre es auch okay, finde ich: wenn ein savedAttribute == null ist und der newValue truesy ist, sollte changed auch true sein. Aber ich glaube du kannst daraus auch === machen, ohne das es akute Auswirkungen hat weil der null-Fall nicht vorkommt und selbst wenn ist ja okay, dass er dann in die else geht und mit isEqual die beiden Werte vergleicht.

Reply:
Na gut, bei clarat passiert das vielleicht (noch) nicht, aber man nehme mal folgenden konstruierten Fall: Es gibt ein Feld einer Altersbeschränkung, welches beim Create-Fall ausdrücklich als nicht vorhanden (`null`) angegeben wurde. Im edit wird das geladen und in die seedData gegeben, dann aber zu der Zahl `0` verändert. Ist in JS falsy, daher würde er das dann nicht als Änderung erkennen, oder? Wobei das Problem auch besteht, wenn es vorher undefined war, also die Daten gar nicht erst in die seedData reingegeben werden.

Reply:
Okay aber dann müsste es mit === doch passen, oder? Er läuft in den else, lässt isEqual darüber laufen und erkennt es als Änderung.

Reply:
Achja, stimmt